pkg/cloud/digitalocean/provider.go

```diff
@@ -233,14 +233,45 @@ type SlugBase struct {
 	BaseRAMGiB int
 }
 
-// Base slugs and known pricing
-var baseSlugs = map[string]SlugBase{
-	"c": {BaseSlug: "c-4-intel", BaseCost: 0.1622, BaseVCPU: 4, BaseRAMGiB: 8},
-	"s": {BaseSlug: "s-4vcpu-8gb", BaseCost: 0.07143, BaseVCPU: 4, BaseRAMGiB: 8},
-	"m": {BaseSlug: "m-8vcpu-64gb", BaseCost: 0.50, BaseVCPU: 8, BaseRAMGiB: 64},
-	"g": {BaseSlug: "g-4vcpu-16gb-intel", BaseCost: 0.2247, BaseVCPU: 4, BaseRAMGiB: 16},
+type slugSeeds struct {
+	BaseVCPU    int
+	BaseHourly  float64
+	RamPerVCPU  int
+	IntelHourly float64
+}
+
+var slugFamilySeed = map[string]slugSeeds{
+	"c":     {BaseVCPU: 4, BaseHourly: 0.12500, RamPerVCPU: 2, IntelHourly: 0.16220},
+	"c2":    {BaseVCPU: 4, BaseHourly: 0.13988, RamPerVCPU: 2, IntelHourly: 0.18155},
+	"g":     {BaseVCPU: 4, BaseHourly: 0.18750, RamPerVCPU: 4, IntelHourly: 0.22470},
+	"gd":    {BaseVCPU: 4, BaseHourly: 0.20238, RamPerVCPU: 4, IntelHourly: 0.23512},
+	"m":     {BaseVCPU: 8, BaseHourly: 0.50000, RamPerVCPU: 8, IntelHourly: 0.58929},
+	"m3":    {BaseVCPU: 8, BaseHourly: 0.61905, RamPerVCPU: 8, IntelHourly: 0.65476},
+	"m6":    {BaseVCPU: 8, BaseHourly: 0.77976, RamPerVCPU: 8, IntelHourly: 0},
+	"s":     {BaseVCPU: 4, BaseHourly: 0.07143, RamPerVCPU: 2, IntelHourly: 0.08333},
+	"so":    {BaseVCPU: 8, BaseHourly: 0.77976, RamPerVCPU: 8, IntelHourly: 0.77976},
+	"so1_5": {BaseVCPU: 8, BaseHourly: 0.97024, RamPerVCPU: 8, IntelHourly: 0.82738},
+}
```

Comment:
@prateek1192 will these go out of date as DO changes its pricing?

Reply:
Thanks for pointing this out Alex and internally in DO as well we have had some discussion about this. DO raises prices infrequently(only one since launch since 2011) and this is a fallback and we also majorly plan on relying on the endpoint that is being built for billing. So we decided to keep it like this since this was a trade off of updating pricing every 10 years to having all nodes accurate prices here.
I looked at other providers and saw that they can add CPU and RAM prices using a config file. The problem with that for DO is that we have different CPU and RAM prices for different slugs(node types) and don't have one size fits all model.